### ISSUE TITLE
build_dataset fails when aggregating timestamps into buckets

### ISSUE THREAD

**Issue by juancq** [mmcdermott/EventStreamGPT#119]
Branch: dev 
build_dataset runs out of memory when aggregating timestamps into buckets. 

```fish
2024-08-23 13:21:58.081 | DEBUG    | EventStream.data.dataset_base:__init__:479 - Built events and measurements dataframe
2024-08-23 13:21:58.085 | DEBUG    | EventStream.data.dataset_polars:_agg_by_time:642 - Collecting events DF. Not using streaming here as it sometimes causes segfaults.
2024-08-23 13:22:06.849 | DEBUG    | EventStream.data.dataset_polars:_agg_by_time:649 - Aggregating timestamps into buckets
fish: Job 1, 'PYTHONPATH=".;"  python scripts…' terminated by signal SIGKILL (Forced quit)
```
The relevant code is this:
https://github.com/mmcdermott/EventStreamGPT/blob/ce9e2c3e80c00a79ff6ee53deaee9ec6ca6f2669/EventStream/data/dataset_polars.py#L648-L676

To replicate, run generate_synthetic_data with n_subjects > 50,000 and then run build_dataset. 

The solution with my dataset (about 5 million subjects) was to use a compute instance with more memory during the build_dataset phase. 

This is partly a polars issue. I tried limiting number of threads and streaming, and it did not make a difference. 

A refactor of agg_by_time would be nice to have, but not a must.

**Comment by mmcdermott** [comment]
Yes, it has been found that ESGPT as written doesn't scale in memory well as you approach billions of clinical events. The [MEDS standard](https://medical-event-data-standard.github.io/) supports this much better, so as we finalize the upcoming deprecation of ESGPT towards MEDS and provide dedicated transition code, that should help improve this issue. For now, I'm closing this issue as we don't plan to make any modifications to the code for memory efficiency in this library given the shift to MEDS.
